solc: 0.4.25 -> 0.5.0

asymmetric opened the pull request:
###### Motivation for this change

Updated to newest version.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

asymmetric committed:
solc: 0.4.25 -> 0.5.0

Mic92 commented:
@GrahamcOfBorg build solc

GrahamcOfBorg commented:
<!--REQUEST_ID=59b3a35e-7c9d-43c6-b3dc-d3469f660c03-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.50324&attempt_id=844b1e2b-c38e-4218-b9fd-c29a7af25edf)

Attempted: solc

<details><summary>Partial log (click to expand)</summary><p>

```
 #include <immintrin.h>
          ^~~~~~~~~~~~~
compilation terminated.
src/sat/sat_probing.cpp
make: *** [Makefile:182: util/mpz.o] Error 1
make: *** Waiting for unfinished jobs....
src/sat/sat_parallel.cpp
builder for '/nix/store/vd2wkhic8g77izxv659ackh6hcaamic3-z3-4.8.1.drv' failed with exit code 2
cannot build derivation '/nix/store/h5imnjsf31c45l558gw66vyzb0ickc1m-solc-0.5.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/h5imnjsf31c45l558gw66vyzb0ickc1m-solc-0.5.0.drv' failed
```
</p></details>

Mic92 committed:
z3: restrict to x86_64

it uses x86-only header:

include <immintrin.h>
          ^~~~~~~~~~~~~
compilation terminated.
src/sat/sat_probing.cpp
make: *** [Makefile:182: util/mpz.o] Error 1
make: *** Waiting for unfinished jobs....
src/sat/sat_parallel.cpp
builder for '/nix/store/vd2wkhic8g77izxv659ackh6hcaamic3-z3-4.8.1.drv' failed with exit code 2
cannot build derivation '/nix/store/h5imnjsf31c45l558gw66vyzb0ickc1m-solc-0.5.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/h5imnjsf31c45l558gw66vyzb0ickc1m-solc-0.5.0.drv' failed

GrahamcOfBorg commented:
<!--REQUEST_ID=59b3a35e-7c9d-43c6-b3dc-d3469f660c03-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.50324&attempt_id=634369bc-40c1-4b36-ac67-76b2f55f1ccb)

Attempted: solc

<details><summary>Partial log (click to expand)</summary><p>

```
-- Installing: /nix/store/yzzmqzrj020v46dqlra0c3h4xiifr3kb-solc-0.5.0/lib/libsolidity.dylib
-- Installing: /nix/store/yzzmqzrj020v46dqlra0c3h4xiifr3kb-solc-0.5.0/bin/solc
-- Installing: /nix/store/yzzmqzrj020v46dqlra0c3h4xiifr3kb-solc-0.5.0/lib/liblll.dylib
-- Installing: /nix/store/yzzmqzrj020v46dqlra0c3h4xiifr3kb-solc-0.5.0/bin/lllc
post-installation fixup
strip is /nix/store/g5r4apl0za012ffs6ladinwa5w0m1l3k-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/yzzmqzrj020v46dqlra0c3h4xiifr3kb-solc-0.5.0/lib  /nix/store/yzzmqzrj020v46dqlra0c3h4xiifr3kb-solc-0.5.0/bin
patching script interpreter paths in /nix/store/yzzmqzrj020v46dqlra0c3h4xiifr3kb-solc-0.5.0
strip is /nix/store/g5r4apl0za012ffs6ladinwa5w0m1l3k-cctools-binutils-darwin/bin/strip
patching script interpreter paths in /nix/store/48f1y9qkkmm3d7byb1zr1d2rlx07v0rm-solc-0.5.0-dev
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=59b3a35e-7c9d-43c6-b3dc-d3469f660c03-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.50324&attempt_id=1d8f0acb-cb7c-419d-923b-ce2e72a90363)

Attempted: solc

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/6dxyvp5k929kag0xvz44jp23vari934q-solc-0.5.0/bin/lllc
strip is /nix/store/vcc4svb8gy29g4pam2zja6llkbcwsyiq-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/6dxyvp5k929kag0xvz44jp23vari934q-solc-0.5.0/lib  /nix/store/6dxyvp5k929kag0xvz44jp23vari934q-solc-0.5.0/bin
patching script interpreter paths in /nix/store/6dxyvp5k929kag0xvz44jp23vari934q-solc-0.5.0
checking for references to /build in /nix/store/6dxyvp5k929kag0xvz44jp23vari934q-solc-0.5.0...
shrinking RPATHs of ELF executables and libraries in /nix/store/paj376wsb1zrnf4kl03lhm6xb4fzr7zq-solc-0.5.0-dev
strip is /nix/store/vcc4svb8gy29g4pam2zja6llkbcwsyiq-binutils-2.30/bin/strip
patching script interpreter paths in /nix/store/paj376wsb1zrnf4kl03lhm6xb4fzr7zq-solc-0.5.0-dev
checking for references to /build in /nix/store/paj376wsb1zrnf4kl03lhm6xb4fzr7zq-solc-0.5.0-dev...
/nix/store/6dxyvp5k929kag0xvz44jp23vari934q-solc-0.5.0
```
</p></details>
